### PR TITLE
CORS-3302: port forward SSH to bootstrap host

### PR DESCRIPTION
**Pull request by jhixson74**
Depends on https://github.com/openshift/installer/pull/8652

https://issues.redhat.com/browse/CORS-3302

### CODE REVIEW

File: pkg/infrastructure/azure/azure.go [hunk 1]
```diff
@@ -471,7 +476,9 @@ func (p *Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput
 		}
 		logrus.Debugf("created public ip: %s", *publicIP.ID)
 
-		loadBalancer, err := updateExternalLoadBalancer(ctx, publicIP, lbInput)
+		// TODO: spawn thread to wait for machine creation and
+		// associate to new backend pool.
```
Comment (patrickdillon) [inline pkg/infrastructure/azure/azure.go:480]: I think this is done (in a different way) already, right? if so we just need to delete the comment

**Review by patrickdillon — COMMENTED** [review]
Cool. This LGTM

/approve

Additionally, we will need to extend our providers interface to allow a customization of [ExtractHostAddresses](https://github.com/openshift/installer/blob/9ecbd30afbc7a2f7b3be44f9e9bf6c9671b21582/pkg/infrastructure/clusterapi/clusterapi.go#L480) and use that to grab the load balancer ip address for the bootstrap node. But we can handle that separately, should just capture it as a card or somewhere.

File: pkg/infrastructure/azure/azure.go (hunk 1 above)
Comment (jhixson74) [reply pkg/infrastructure/azure/azure.go:480]: They are eventually joined, yes. Unfortunately, this delays the install time a bit. The new backend pool is added in InfraReady() while the join doesn't happen until PostProvision(). Ideally we want machines added to the pool as they are provisioned.

File: pkg/infrastructure/azure/network.go [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+
+	interfacesResp, err := interfacesPollerResp.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to add inbound nat rule to interface: %w", err)
+	}
+	return &interfacesResp.Interface, nil
+}
+
+// GetCloudConfiguration gets a cloud configuration from the cloud name and endpoint.
+func GetCloudConfiguration(cloudName aztypes.CloudEnvironment, armEndpoint string) (*cloud.Configuration, error) {
```
Comment (r4f4) [inline pkg/infrastructure/azure/network.go:439]: For the future, this should be either grabbed from the session: https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/azure/session.go#L86-L104 + https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/azure/session.go#L133 or the code should be shared by both.

File: pkg/infrastructure/azure/network.go [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+					Endpoint: cloudEnv.ResourceManagerEndpoint,
+				},
+			},
+		}
+	case aztypes.USGovernmentCloud:
+		cloudConfig = cloud.AzureGovernment
+	case aztypes.ChinaCloud:
+		cloudConfig = cloud.AzureChina
+	// XXX: What do we do here?
+	// case aztypes.GermanCloud:
```
Comment (r4f4) [inline pkg/infrastructure/azure/network.go:462]: I don't think we need to worry about this as the German cloud was shutdown: https://github.com/Azure/azure-sdk-for-go/issues/20293#issuecomment-1446485961, https://www.azurespeed.com/Information/AzureSovereignClouds

File: pkg/infrastructure/azure/azure.go (hunk 1 above)
Comment (jhixson74) [reply pkg/infrastructure/azure/azure.go]: Removed the comment.

File: pkg/infrastructure/azure/network.go (hunk 2 above)
Comment (jhixson74) [reply pkg/infrastructure/azure/network.go]: Fixed.

File: pkg/infrastructure/azure/network.go (hunk 3 above)
Comment (jhixson74) [reply pkg/infrastructure/azure/network.go]: Fixed.

**Comment by jhixson74** [comment]
> `golint` tests has legit failures: https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_installer/8669/pull-ci-openshift-installer-master-golint/1806786170038784000

Fixed.

File: pkg/asset/installconfig/azure/session.go [hunk 4]
```diff
@@ -130,11 +115,41 @@ func GetSessionWithCredentials(cloudName azure.CloudEnvironment, armEndpoint str
...
 	}
-	session.CloudConfig = cloudConfig
+	session.CloudConfig = *cloudConfig
 	session.AuthType = authType
 	return session, nil
 }
 
+// GetCloudConfiguration gets a cloud configuration from the cloud name and endpoint.
+func GetCloudConfiguration(cloudName azure.CloudEnvironment, armEndpoint string) (*cloud.Configuration, error) {
+	cloudEnv, err := azureenv.EnvironmentFromURL(armEndpoint)
```
Comment (r4f4) [inline pkg/asset/installconfig/azure/session.go:125]: This is missing the non-stackcloud case https://github.com/openshift/installer/pull/8669/files#diff-da950a4ddabbede621d9d3b1058bb34f8931c89179306ee88a0e4d76a4cf0b13R74-R81.

This is causing the failure in https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_installer/8669/pull-ci-openshift-installer-master-e2e-azure-ovn/1811162760466141184

**Review by r4f4 — COMMENTED** [review]
/lgtm

**Comment by r4f4** [comment]
/hold
because of the dep on https://github.com/openshift/installer/pull/8652

**Comment by jhixson74** [comment]
/hold cancel

#8652 has merged

**Pull request merged**